org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/UpdateClasspathJob.java

```diff
@@ -58,24 +61,25 @@ public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
 			requests = new ArrayList<>(this.queue);
 			this.queue.clear();
 		}
-		Map<IJavaProject, Optional<UpdateClasspathRequest>> mergedRequestPerProject = requests.stream().collect(
+		Map<IJavaProject, UpdateClasspathRequest> mergedRequestPerProject = requests.stream().collect(
 			Collectors.groupingBy(UpdateClasspathRequest::getProject,
-				Collectors.reducing((mergedRequest, request) -> {
+				Collectors.reducing(new UpdateClasspathRequest(), (mergedRequest, request) -> {
```

Comment:
An empty request is constructed for merged request, in order not to modify collections in requests from queue.